ctest.h

```diff
@@ -79,14 +79,19 @@ CTEST_IMPL_DIAG_POP()
 #define CTEST_IMPL_TEARDOWN_FPNAME(sname) CTEST_IMPL_NAME(sname##_teardown_ptr)
 
 #define CTEST_IMPL_MAGIC (0xdeadbeef)
-#ifdef __APPLE__
+#if defined(_MSC_VER)
+#pragma data_seg(push)
+#pragma data_seg(".ctest$u")
+#pragma data_seg(pop)
+#define CTEST_IMPL_SECTION __declspec(allocate(".ctest$u")) __declspec(align(1))
+#elif defined(__APPLE__)
 #define CTEST_IMPL_SECTION __attribute__ ((used, section ("__DATA, .ctest"), aligned(1)))
```

Comment:
I didn't find anything like `used` right away. Also don't recall whether I tested the optimized build or not, will need to get back to you on this.

Reply:
A simple and portable workaround for the symbol-stripping problem would be to remove `static` from the definitions. Compilers cannot strip non-static symbols even if they appear unused since another compilation unit may be `extern`ing them. This would also remove the need for the `used` attribute in the `__GNUC__` case.